src/System.Private.CoreLib/shared/System/Collections/ObjectModel/Collection.cs

```diff
@@ -372,16 +373,16 @@ void ICollection.CopyTo(Array array, int index)
             }
         }
 
-        object IList.this[int index]
+        object? IList.this[int index]
         {
             get { return items[index]; }
             set
             {
-                ThrowHelper.IfNullAndNullsAreIllegalThenThrow<T>(value, ExceptionArgument.value);
+                ThrowHelper.IfNullAndNullsAreIllegalThenThrow<T>(value!, ExceptionArgument.value);
```

Comment:
We should probably make `IfNullAndNullsAreIllegalThenThrow` take an `object?` as its first argument.

Reply:
Reading further in the PR, I see you did that... why is the `!` needed here then?